### DISCUSSION TITLE
Initializing databse ASAP

### DISCUSSION THREAD

**Discussion by Abbos27** [simolus3/drift#3050] [Q&A] [answered]
I want to initialize my database as soon as possible, not lazily. Can I directly call NativeDatabase.createInBackground(file) and pass it to the super constructor of GeneratedDatabase?

**Comment by simolus3** [comment, marked as answer]
`NativeDatabase.createInBackground` also uses another `LazyDatabase` internally - it has to because it needs to spawn an isolate and setup a communication channel with it to actually use the database. The most reliable way to initialize the database right away is to use it, e.g. like this:

```dart
Future<YourDatabaseClass> loadInitializedDatabase() async {
  final database = YourDatabaseClass(LazyDatabase(...));
  await database.customSelect('SELECT 1').get(); // run a statement, forcing the database to be initialized
  return database;
}
```
